common_clang.cpp

```diff
@@ -235,15 +235,13 @@ Compile(const char *pszProgramSource, const char **pInputHeaders,
 
     compiler->setDiagnostics(&*Diags);
 
-    auto OverlayFS = new llvm::vfs::OverlayFileSystem(
-        llvm::vfs::getRealFileSystem());
-    auto MemFS = new llvm::vfs::InMemoryFileSystem();
+    llvm::IntrusiveRefCntPtr<llvm::vfs::OverlayFileSystem> OverlayFS(
+        new llvm::vfs::OverlayFileSystem(llvm::vfs::getRealFileSystem()));
+    llvm::IntrusiveRefCntPtr<llvm::vfs::InMemoryFileSystem> MemFS(
+        new llvm::vfs::InMemoryFileSystem);
     OverlayFS->pushOverlay(MemFS);
 
-    compiler->setVirtualFileSystem(OverlayFS);
-
-    compiler->createFileManager();
-    compiler->createSourceManager(compiler->getFileManager());
+    compiler->createSourceManager(*compiler->createFileManager(OverlayFS));
```

Comment:
What if `createFileManager` return `nullptr`?

Reply:
I returned it as it was before. So assert should fire in getFileManager in case the pointer is zero.